From: Janne Grunau <j@jannau.net>
To: Lukasz Majewski <lukma@denx.de>, u-boot@lists.denx.de
Cc: AKASHI Takahiro <takahiro.akashi@linaro.org>, Simon Glass
 <sjg@chromium.org>
Subject: [PATCH v2 1/1] usb: storage: continue probe on "Invalid device"
Date: Wed, 10 Aug 2022 21:54:22 +0200
X-Mailer: git-send-email 2.35.1
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Content-Type: text/plain; charset=utf-8

Fixes a crash during probing of sd card readers without medium present.

Link: https://github.com/AsahiLinux/linux/issues/44
Link: https://lists.denx.de/pipermail/u-boot/2022-July/489717.html
Signed-off-by: Janne Grunau <j@jannau.net>
---
Changes since v1:
 - changed unconditiona return to "continue" as proposed by AKASHI Takahiro

 common/usb_storage.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/usb_storage.c b/common/usb_storage.c
index eaa31374ef73..f9204552a683 100644
--- a/common/usb_storage.c
+++ b/common/usb_storage.c
@@ -239,6 +239,7 @@ static int usb_stor_probe_device(struct usb_device *udev)
 			ret = device_unbind(dev);
 			if (ret)
 				return ret;
+			continue;
 		}
 
 		ret = blk_probe_or_unbind(dev);
-- 
2.35.1

